packages/legacy-client/src/client/interfaces/aggregations.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { LocalDate, Timestamp } from "../..";
+import type {
+  AggregatableProperty,
+  ApproximateDistinctCountAggregatableProperty,
+  BooleanGroupBy,
+  Bucketing,
+  BucketKey,
+  BucketValue,
+  CountOperation,
+  Double,
+  LocalDateGroupBy,
+  MultipleAggregatableProperty,
+  NumericGroupBy,
+  OntologyObject,
+  StringGroupBy,
+  TimestampGroupBy,
+} from "../../ontology-runtime";
+import type { OmitMetadataProperties } from "./utils/OmitProperties";
+
+export declare type ObjectTypesGroupByFunction<
+  T extends OntologyObject,
+  K extends BucketKey,
+  V extends BucketValue,
+> = (
+  objectType: GroupBySelections<T>,
+) => Bucketing<K, V>;
+
+type IsGroupableProperty<T> = NonNullable<T> extends
+  number | LocalDate | Timestamp | string | boolean ? true : false;
+
+type GroupableProperties<T extends OntologyObject> = {
+  [K in keyof T as IsGroupableProperty<T[K]> extends true ? K : never]: T[K];
+};
+
+export declare type GroupBySelections<T extends OntologyObject> = {
```

Comment:
I need to push my PR so you can see but this probably should be `<O extends OntologyDefinition<O>, K extends OntologyTypesFrom<O>>`. I've got a couple examples on my local for ObjectSet and friends